### PR TITLE
Adding test to check fro PF eval dependencies on private API's

### PR DESCRIPTION
**Pull request by singankit**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by nick863 — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_sdk/_pf_evals_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# This file captures promptflow-evals dependencies on private API of promptflow.
+# In case changes are made please reach out to promptflow-evals team to update the dependencies.
+
+
+def _get_pf_evals_dependencies():
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_pf_evals_dependencies.py:9]: why do we need a function here? why not directly have some imports here?

Comment (singankit) [reply src/promptflow-devkit/promptflow/_sdk/_pf_evals_dependencies.py:9]: Using this method in test to make sure imports are not broken. If file is not loaded then imports command might not run. Open to suggestions if this approach is not full proof.

File: src/promptflow-azure/promptflow/azure/_pf_eval_dependencies.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# ---------------------------------------------------------
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/_pf_eval_dependencies.py:1]: Maybe orgnize it like: so we can keep the root namepsace cleaner
_dependencies/_pf_eval.py

File: src/promptflow-devkit/promptflow/_sdk/_pf_evals_dependencies.py (hunk 1 above)
Comment (0mza987) [reply src/promptflow-devkit/promptflow/_sdk/_pf_evals_dependencies.py:9]: Think clement means you can create a folder which has `__init__.py` inside and import all private api there, and whenever you need these private apis, just import from this init file. Here is a practice you can check out:
[imports_from_azure_ai_ml.md - Repos](https://msdata.visualstudio.com/Vienna/_git/sdk-cli-v2?path=%2Fdocs%2Fmldesigner%2Fimports_from_azure_ai_ml.md&_a=preview)

Comment (singankit) [reply src/promptflow-devkit/promptflow/_sdk/_pf_evals_dependencies.py]: Got pretty close to it. Have imported dependencies in file and then using them in test.

File: src/promptflow-devkit/promptflow/_dependencies/_pf_evals.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# This file captures promptflow-evals dependencies on private API of promptflow.
+# In case changes are made please reach out to promptflow-evals team to update the dependencies.
+
+from promptflow._sdk._constants import LINE_NUMBER, Local2Cloud
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_dependencies/_pf_evals.py:8]: turns out this will raise flake8 error

Comment (singankit) [reply src/promptflow-devkit/promptflow/_dependencies/_pf_evals.py:8]: Shall I go ahead and ignore the flake8 error

**Review by wangchao1230 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**